File: src/pycropml/transpiler/antlr_py/csharp/examples/pheno_pkg/src/cs/PhenologyComponent.cs
public class PhenologyComponent
{
    
    public PhenologyComponent() { }
    

    //Declaration of the associated strategies
    Phyllochron _Phyllochron = new Phyllochron();
    PhylSowingDateCorrection _PhylSowingDateCorrection = new PhylSowingDateCorrection();
    ShootNumber _ShootNumber = new ShootNumber();
    UpdateLeafFlag _UpdateLeafFlag = new UpdateLeafFlag();
    UpdatePhase _UpdatePhase = new UpdatePhase();
    LeafNumber _LeafNumber = new LeafNumber();
    VernalizationProgress _VernalizationProgress = new VernalizationProgress();
    IsMomentRegistredZC_39 _IsMomentRegistredZC_39 = new IsMomentRegistredZC_39();
    CumulTTFrom _CumulTTFrom = new CumulTTFrom();
    UpdateCalendar _UpdateCalendar = new UpdateCalendar();
    RegisterZadok _RegisterZadok = new RegisterZadok();
    PTQ _PTQ = new PTQ();
    GAImean _GAImean = new GAImean();

    public double aMXLFNO
    {
        get
        {
             return _VernalizationProgress.aMXLFNO; 
        }
        set
        {
            _VernalizationProgress.aMXLFNO = value;
        }
    }
    public double pNini
    {
        get
        {
             return _VernalizationProgress.pNini; 
        }
        set
        {
            _VernalizationProgress.pNini = value;
        }
    }
    public double sDsa_sh
    {
        get
        {
             return _PhylSowingDateCorrection.sDsa_sh; 
        }
        set
        {
            _PhylSowingDateCorrection.sDsa_sh = value;
        }
    }
    public double latitude
    {
        get
        {
             return _PhylSowingDateCorrection.latitude; 
        }
        set
        {
            _PhylSowingDateCorrection.latitude = value;
        }
    }
    public double kl
    {
        get
        {
             return _Phyllochron.kl; 
        }
        set
        {
            _Phyllochron.kl = value;
            _PTQ.kl = value;
        }
    }
    public double lincr
    {
        get
        {
             return _Phyllochron.lincr; 
        }
        set
        {
            _Phyllochron.lincr = value;
        }
    }
    public double ldecr
    {
        get
        {
             return _Phyllochron.ldecr; 
        }
        set
        {
            _Phyllochron.ldecr = value;
        }
    }
    public double pincr
    {
        get
        {
             return _Phyllochron.pincr; 
        }
        set
        {
            _Phyllochron.pincr = value;
        }
    }
    public double pTQhf
    {
        get
        {
             return _Phyllochron.pTQhf; 
        }
        set
        {
            _Phyllochron.pTQhf = value;
        }
    }
    public double B
    {
        get
        {
             return _Phyllochron.B; 
        }
        set
        {
            _Phyllochron.B = value;
        }
    }
    public double areaSL
    {
        get
        {
             return _Phyllochron.areaSL; 
        }
        set
        {
            _Phyllochron.areaSL = value;
        }
    }
    public double areaSS
    {
        get
        {
             return _Phyllochron.areaSS; 
        }
        set
        {
            _Phyllochron.areaSS = value;
        }
    }
    public double lARmin
    {
        get
        {
             return _Phyllochron.lARmin; 
        }
        set
        {
            _Phyllochron.lARmin = value;
        }
    }
    public double sowingDensity
    {
        get
        {
             return _Phyllochron.sowingDensity; 
        }
        set
        {
            _Phyllochron.sowingDensity = value;
            _ShootNumber.sowingDensity = value;
        }
    }
    public double lARmax
    {
        get
        {
             return _Phyllochron.lARmax; 
        }
        set
        {
            _Phyllochron.lARmax = value;
        }
    }
    public double lNeff
    {
        get
        {
             return _Phyllochron.lNeff; 
        }
        set
        {
            _Phyllochron.lNeff = value;
        }
    }
    public double rp
    {
        get
        {
             return _PhylSowingDateCorrection.rp; 
        }
        set
        {
            _PhylSowingDateCorrection.rp = value;
        }
    }
    public double p
    {
        get
        {
             return _Phyllochron.p; 
        }
        set
        {
            _Phyllochron.p = value;
            _PhylSowingDateCorrection.p = value;
            _UpdatePhase.p = value;
        }
    }
    public double pdecr
    {
        get
        {
             return _Phyllochron.pdecr; 
        }
        set
        {
            _Phyllochron.pdecr = value;
        }
    }
    public double maxTvern
    {
        get
        {
             return _VernalizationProgress.maxTvern; 
        }
        set
        {
            _VernalizationProgress.maxTvern = value;
        }
    }
    public double tTWindowForPTQ
    {
        get
        {
             return _GAImean.tTWindowForPTQ; 
        }
        set
        {
            _GAImean.tTWindowForPTQ = value;
            _PTQ.tTWindowForPTQ = value;
        }
    }
    public double vBEE
    {
        get
        {
             return _VernalizationProgress.vBEE; 
        }
        set
        {
            _VernalizationProgress.vBEE = value;
        }
    }
    public int isVernalizable
    {
        get
        {
             return _VernalizationProgress.isVernalizable; 
        }
        set
        {
            _VernalizationProgress.isVernalizable = value;
            _UpdatePhase.isVernalizable = value;
        }
    }
    public double minTvern
    {
        get
        {
             return _VernalizationProgress.minTvern; 
        }
        set
        {
            _VernalizationProgress.minTvern = value;
        }
    }
    public double intTvern
    {
        get
        {
             return _VernalizationProgress.intTvern; 
        }
        set
        {
            _VernalizationProgress.intTvern = value;
        }
    }
    public double vAI
    {
        get
        {
             return _VernalizationProgress.vAI; 
        }
        set
        {
            _VernalizationProgress.vAI = value;
        }
    }
    public double maxDL
    {
        get
        {
             return _VernalizationProgress.maxDL; 
        }
        set
        {
            _VernalizationProgress.maxDL = value;
            _UpdatePhase.maxDL = value;
        }
    }
    public string choosePhyllUse
    {
        get
        {
             return _Phyllochron.choosePhyllUse; 
        }
        set
        {
            _Phyllochron.choosePhyllUse = value;
            _UpdatePhase.choosePhyllUse = value;
        }
    }
    public double minDL
    {
        get
        {
             return _VernalizationProgress.minDL; 
        }
        set
        {
            _VernalizationProgress.minDL = value;
        }
    }
    public double pFLLAnth
    {
        get
        {
             return _UpdatePhase.pFLLAnth; 
        }
        set
        {
            _UpdatePhase.pFLLAnth = value;
        }
    }
    public double dcd
    {
        get
        {
             return _UpdatePhase.dcd; 
        }
        set
        {
            _UpdatePhase.dcd = value;
        }
    }
    public double dgf
    {
        get
        {
             return _UpdatePhase.dgf; 
        }
        set
        {
            _UpdatePhase.dgf = value;
        }
    }
    public double degfm
    {
        get
        {
             return _UpdatePhase.degfm; 
        }
        set
        {
            _UpdatePhase.degfm = value;
        }
    }
    public bool ignoreGrainMaturation
    {
        get
        {
             return _UpdatePhase.ignoreGrainMaturation; 
        }
        set
        {
            _UpdatePhase.ignoreGrainMaturation = value;
        }
    }
    public double pHEADANTH
    {
        get
        {
             return _UpdatePhase.pHEADANTH; 
        }
        set
        {
            _UpdatePhase.pHEADANTH = value;
        }
    }
    public double sLDL
    {
        get
        {
             return _UpdatePhase.sLDL; 
        }
        set
        {
            _UpdatePhase.sLDL = value;
        }
    }
    public int sowingDay
    {
        get
        {
             return _PhylSowingDateCorrection.sowingDay; 
        }
        set
        {
            _PhylSowingDateCorrection.sowingDay = value;
        }
    }
    public int sDws
    {
        get
        {
             return _PhylSowingDateCorrection.sDws; 
        }
        set
        {
            _PhylSowingDateCorrection.sDws = value;
        }
    }
    public double sDsa_nh
    {
        get
        {
             return _PhylSowingDateCorrection.sDsa_nh; 
        }
        set
        {
            _PhylSowingDateCorrection.sDsa_nh = value;
        }
    }
    public double der
    {
        get
        {
             return _RegisterZadok.der; 
        }
        set
        {
            _RegisterZadok.der = value;
        }
    }
    public double targetFertileShoot
    {
        get
        {
             return _ShootNumber.targetFertileShoot; 
        }
        set
        {
            _ShootNumber.targetFertileShoot = value;
        }
    }
    public double dse
    {
        get
        {
             return _UpdatePhase.dse; 
        }
        set
        {
            _UpdatePhase.dse = value;
        }
    }
    public double slopeTSFLN
    {
        get
        {
             return _RegisterZadok.slopeTSFLN; 
        }
        set
        {
            _RegisterZadok.slopeTSFLN = value;
        }
    }
    public double intTSFLN
    {
        get
        {
             return _RegisterZadok.intTSFLN; 
        }
        set
        {
            _RegisterZadok.intTSFLN = value;
        }
    }

    public void  CalculateModel(PhenologyState s, PhenologyState s1, PhenologyRate r, PhenologyAuxiliary a)
    {
        _GAImean.CalculateModel(s,s1, r, a);
        _PTQ.CalculateModel(s,s1, r, a);
        _CumulTTFrom.CalculateModel(s,s1, r, a);
        _IsMomentRegistredZC_39.CalculateModel(s,s1, r, a);
        _VernalizationProgress.CalculateModel(s,s1, r, a);
        _PhylSowingDateCorrection.CalculateModel(s,s1, r, a);
        _UpdatePhase.CalculateModel(s,s1, r, a);
        _LeafNumber.CalculateModel(s,s1, r, a);
        _ShootNumber.CalculateModel(s,s1, r, a);
        _UpdateLeafFlag.CalculateModel(s,s1, r, a);
        _RegisterZadok.CalculateModel(s,s1, r, a);
        _UpdateCalendar.CalculateModel(s,s1, r, a);
        _Phyllochron.CalculateModel(s,s1, r, a);
    }
    
    public PhenologyComponent(PhenologyComponent toCopy): this() // copy constructor 
    {

        aMXLFNO = toCopy.aMXLFNO;
        pNini = toCopy.pNini;
        sDsa_sh = toCopy.sDsa_sh;
        latitude = toCopy.latitude;
        kl = toCopy.kl;
        lincr = toCopy.lincr;
        ldecr = toCopy.ldecr;
        pincr = toCopy.pincr;
        pTQhf = toCopy.pTQhf;
        B = toCopy.B;
        areaSL = toCopy.areaSL;
        areaSS = toCopy.areaSS;
        lARmin = toCopy.lARmin;
        sowingDensity = toCopy.sowingDensity;
        lARmax = toCopy.lARmax;
        lNeff = toCopy.lNeff;
        rp = toCopy.rp;
        p = toCopy.p;
        pdecr = toCopy.pdecr;
        maxTvern = toCopy.maxTvern;
        tTWindowForPTQ = toCopy.tTWindowForPTQ;
        vBEE = toCopy.vBEE;
        isVernalizable = toCopy.isVernalizable;
        minTvern = toCopy.minTvern;
        intTvern = toCopy.intTvern;
        vAI = toCopy.vAI;
        maxDL = toCopy.maxDL;
        choosePhyllUse = toCopy.choosePhyllUse;
        minDL = toCopy.minDL;
        pFLLAnth = toCopy.pFLLAnth;
        dcd = toCopy.dcd;
        dgf = toCopy.dgf;
        degfm = toCopy.degfm;
        ignoreGrainMaturation = toCopy.ignoreGrainMaturation;
        pHEADANTH = toCopy.pHEADANTH;
        sLDL = toCopy.sLDL;
        sowingDay = toCopy.sowingDay;
        sDws = toCopy.sDws;
        sDsa_nh = toCopy.sDsa_nh;
        der = toCopy.der;
        targetFertileShoot = toCopy.targetFertileShoot;
        dse = toCopy.dse;
        slopeTSFLN = toCopy.slopeTSFLN;
        intTSFLN = toCopy.intTSFLN;
    }
    

    

    public void Init(PhenologyState s, PhenologyState s1, PhenologyRate r, PhenologyAuxiliary a)
    {
        s.calendarMoments.Add("Sowing");
        s.calendarCumuls.Add(0.0d);
        s.calendarDates.Add(sowingDate);
        s.minFinalNumber = 5.5d;
    }
    
}

File: test/Models/pheno_pkg/src/record/PhenologyComponent.cpp
#include "PhenologyComponent.h"

    PhenologyComponent::PhenologyComponent()
    {
           
    }
    

double PhenologyComponent::getaMXLFNO() {return this-> aMXLFNO; }
double PhenologyComponent::getpNini() {return this-> pNini; }
double PhenologyComponent::getsDsa_sh() {return this-> sDsa_sh; }
double PhenologyComponent::getlatitude() {return this-> latitude; }
double PhenologyComponent::getkl() {return this-> kl; }
double PhenologyComponent::getlincr() {return this-> lincr; }
double PhenologyComponent::getldecr() {return this-> ldecr; }
double PhenologyComponent::getpincr() {return this-> pincr; }
double PhenologyComponent::getpTQhf() {return this-> pTQhf; }
double PhenologyComponent::getB() {return this-> B; }
double PhenologyComponent::getareaSL() {return this-> areaSL; }
double PhenologyComponent::getareaSS() {return this-> areaSS; }
double PhenologyComponent::getlARmin() {return this-> lARmin; }
double PhenologyComponent::getsowingDensity() {return this-> sowingDensity; }
double PhenologyComponent::getlARmax() {return this-> lARmax; }
double PhenologyComponent::getlNeff() {return this-> lNeff; }
double PhenologyComponent::getrp() {return this-> rp; }
double PhenologyComponent::getp() {return this-> p; }
double PhenologyComponent::getpdecr() {return this-> pdecr; }
double PhenologyComponent::getmaxTvern() {return this-> maxTvern; }
double PhenologyComponent::gettTWindowForPTQ() {return this-> tTWindowForPTQ; }
double PhenologyComponent::getvBEE() {return this-> vBEE; }
int PhenologyComponent::getisVernalizable() {return this-> isVernalizable; }
double PhenologyComponent::getminTvern() {return this-> minTvern; }
double PhenologyComponent::getintTvern() {return this-> intTvern; }
double PhenologyComponent::getvAI() {return this-> vAI; }
double PhenologyComponent::getmaxDL() {return this-> maxDL; }
string PhenologyComponent::getchoosePhyllUse() {return this-> choosePhyllUse; }
double PhenologyComponent::getminDL() {return this-> minDL; }
double PhenologyComponent::getpFLLAnth() {return this-> pFLLAnth; }
double PhenologyComponent::getdcd() {return this-> dcd; }
double PhenologyComponent::getdgf() {return this-> dgf; }
double PhenologyComponent::getdegfm() {return this-> degfm; }
bool PhenologyComponent::getignoreGrainMaturation() {return this-> ignoreGrainMaturation; }
double PhenologyComponent::getpHEADANTH() {return this-> pHEADANTH; }
double PhenologyComponent::getsLDL() {return this-> sLDL; }
int PhenologyComponent::getsowingDay() {return this-> sowingDay; }
string PhenologyComponent::getsowingDate() {return this-> sowingDate; }
int PhenologyComponent::getsDws() {return this-> sDws; }
double PhenologyComponent::getsDsa_nh() {return this-> sDsa_nh; }
double PhenologyComponent::getder() {return this-> der; }
double PhenologyComponent::gettargetFertileShoot() {return this-> targetFertileShoot; }
double PhenologyComponent::getdse() {return this-> dse; }
double PhenologyComponent::getslopeTSFLN() {return this-> slopeTSFLN; }
double PhenologyComponent::getintTSFLN() {return this-> intTSFLN; }

void PhenologyComponent::setaMXLFNO(double _aMXLFNO)
{
    _Vernalizationprogress.setaMXLFNO(_aMXLFNO);
}
void PhenologyComponent::setpNini(double _pNini)
{
    _Vernalizationprogress.setpNini(_pNini);
}
void PhenologyComponent::setsDsa_sh(double _sDsa_sh)
{
    _Phylsowingdatecorrection.setsDsa_sh(_sDsa_sh);
}
void PhenologyComponent::setlatitude(double _latitude)
{
    _Phylsowingdatecorrection.setlatitude(_latitude);
}
void PhenologyComponent::setkl(double _kl)
{
    _Phyllochron.setkl(_kl);
    _Ptq.setkl(_kl);
}
void PhenologyComponent::setlincr(double _lincr)
{
    _Phyllochron.setlincr(_lincr);
}
void PhenologyComponent::setldecr(double _ldecr)
{
    _Phyllochron.setldecr(_ldecr);
}
void PhenologyComponent::setpincr(double _pincr)
{
    _Phyllochron.setpincr(_pincr);
}
void PhenologyComponent::setpTQhf(double _pTQhf)
{
    _Phyllochron.setpTQhf(_pTQhf);
}
void PhenologyComponent::setB(double _B)
{
    _Phyllochron.setB(_B);
}
void PhenologyComponent::setareaSL(double _areaSL)
{
    _Phyllochron.setareaSL(_areaSL);
}
void PhenologyComponent::setareaSS(double _areaSS)
{
    _Phyllochron.setareaSS(_areaSS);
}
void PhenologyComponent::setlARmin(double _lARmin)
{
    _Phyllochron.setlARmin(_lARmin);
}
void PhenologyComponent::setsowingDensity(double _sowingDensity)
{
    _Phyllochron.setsowingDensity(_sowingDensity);
    _Shootnumber.setsowingDensity(_sowingDensity);
}
void PhenologyComponent::setlARmax(double _lARmax)
{
    _Phyllochron.setlARmax(_lARmax);
}
void PhenologyComponent::setlNeff(double _lNeff)
{
    _Phyllochron.setlNeff(_lNeff);
}
void PhenologyComponent::setrp(double _rp)
{
    _Phylsowingdatecorrection.setrp(_rp);
}
void PhenologyComponent::setp(double _p)
{
    _Phyllochron.setp(_p);
    _Phylsowingdatecorrection.setp(_p);
    _Updatephase.setp(_p);
}
void PhenologyComponent::setpdecr(double _pdecr)
{
    _Phyllochron.setpdecr(_pdecr);
}
void PhenologyComponent::setmaxTvern(double _maxTvern)
{
    _Vernalizationprogress.setmaxTvern(_maxTvern);
}
void PhenologyComponent::settTWindowForPTQ(double _tTWindowForPTQ)
{
    _Gaimean.settTWindowForPTQ(_tTWindowForPTQ);
    _Ptq.settTWindowForPTQ(_tTWindowForPTQ);
}
void PhenologyComponent::setvBEE(double _vBEE)
{
    _Vernalizationprogress.setvBEE(_vBEE);
}
void PhenologyComponent::setisVernalizable(int _isVernalizable)
{
    _Vernalizationprogress.setisVernalizable(_isVernalizable);
    _Updatephase.setisVernalizable(_isVernalizable);
}
void PhenologyComponent::setminTvern(double _minTvern)
{
    _Vernalizationprogress.setminTvern(_minTvern);
}
void PhenologyComponent::setintTvern(double _intTvern)
{
    _Vernalizationprogress.setintTvern(_intTvern);
}
void PhenologyComponent::setvAI(double _vAI)
{
    _Vernalizationprogress.setvAI(_vAI);
}
void PhenologyComponent::setmaxDL(double _maxDL)
{
    _Vernalizationprogress.setmaxDL(_maxDL);
    _Updatephase.setmaxDL(_maxDL);
}
void PhenologyComponent::setchoosePhyllUse(string _choosePhyllUse)
{
    _Phyllochron.setchoosePhyllUse(_choosePhyllUse);
    _Updatephase.setchoosePhyllUse(_choosePhyllUse);
}
void PhenologyComponent::setminDL(double _minDL)
{
    _Vernalizationprogress.setminDL(_minDL);
}
void PhenologyComponent::setpFLLAnth(double _pFLLAnth)
{
    _Updatephase.setpFLLAnth(_pFLLAnth);
}
void PhenologyComponent::setdcd(double _dcd)
{
    _Updatephase.setdcd(_dcd);
}
void PhenologyComponent::setdgf(double _dgf)
{
    _Updatephase.setdgf(_dgf);
}
void PhenologyComponent::setdegfm(double _degfm)
{
    _Updatephase.setdegfm(_degfm);
}
void PhenologyComponent::setignoreGrainMaturation(bool _ignoreGrainMaturation)
{
    _Updatephase.setignoreGrainMaturation(_ignoreGrainMaturation);
}
void PhenologyComponent::setpHEADANTH(double _pHEADANTH)
{
    _Updatephase.setpHEADANTH(_pHEADANTH);
}
void PhenologyComponent::setsLDL(double _sLDL)
{
    _Updatephase.setsLDL(_sLDL);
}
void PhenologyComponent::setsowingDay(int _sowingDay)
{
    _Phylsowingdatecorrection.setsowingDay(_sowingDay);
}
void PhenologyComponent::setsowingDate(string _sowingDate)
{
    _Registerzadok.setsowingDate(_sowingDate);
}
void PhenologyComponent::setsDws(int _sDws)
{
    _Phylsowingdatecorrection.setsDws(_sDws);
}
void PhenologyComponent::setsDsa_nh(double _sDsa_nh)
{
    _Phylsowingdatecorrection.setsDsa_nh(_sDsa_nh);
}
void PhenologyComponent::setder(double _der)
{
    _Registerzadok.setder(_der);
}
void PhenologyComponent::settargetFertileShoot(double _targetFertileShoot)
{
    _Shootnumber.settargetFertileShoot(_targetFertileShoot);
}
void PhenologyComponent::setdse(double _dse)
{
    _Updatephase.setdse(_dse);
}
void PhenologyComponent::setslopeTSFLN(double _slopeTSFLN)
{
    _Registerzadok.setslopeTSFLN(_slopeTSFLN);
}
void PhenologyComponent::setintTSFLN(double _intTSFLN)
{
    _Registerzadok.setintTSFLN(_intTSFLN);
}
void PhenologyComponent::Calculate_Model(PhenologyState& s, PhenologyState& s1, PhenologyRate& r, PhenologyAuxiliary& a)
{
    _Gaimean.Calculate_Model(s, s1, r, a);
    _Ptq.Calculate_Model(s, s1, r, a);
    _Cumulttfrom.Calculate_Model(s, s1, r, a);
    _Ismomentregistredzc_39.Calculate_Model(s, s1, r, a);
    _Vernalizationprogress.Calculate_Model(s, s1, r, a);
    _Phylsowingdatecorrection.Calculate_Model(s, s1, r, a);
    _Updatephase.Calculate_Model(s, s1, r, a);
    _Leafnumber.Calculate_Model(s, s1, r, a);
    _Shootnumber.Calculate_Model(s, s1, r, a);
    _Updateleafflag.Calculate_Model(s, s1, r, a);
    _Registerzadok.Calculate_Model(s, s1, r, a);
    _Updatecalendar.Calculate_Model(s, s1, r, a);
    _Phyllochron.Calculate_Model(s, s1, r, a);
}
PhenologyComponent::PhenologyComponent(const PhenologyComponent& toCopy)
{
    aMXLFNO = toCopy.aMXLFNO;
    pNini = toCopy.pNini;
    sDsa_sh = toCopy.sDsa_sh;
    latitude = toCopy.latitude;
    kl = toCopy.kl;
    lincr = toCopy.lincr;
    ldecr = toCopy.ldecr;
    pincr = toCopy.pincr;
    pTQhf = toCopy.pTQhf;
    B = toCopy.B;
    areaSL = toCopy.areaSL;
    areaSS = toCopy.areaSS;
    lARmin = toCopy.lARmin;
    sowingDensity = toCopy.sowingDensity;
    lARmax = toCopy.lARmax;
    lNeff = toCopy.lNeff;
    rp = toCopy.rp;
    p = toCopy.p;
    pdecr = toCopy.pdecr;
    maxTvern = toCopy.maxTvern;
    tTWindowForPTQ = toCopy.tTWindowForPTQ;
    vBEE = toCopy.vBEE;
    isVernalizable = toCopy.isVernalizable;
    minTvern = toCopy.minTvern;
    intTvern = toCopy.intTvern;
    vAI = toCopy.vAI;
    maxDL = toCopy.maxDL;
    choosePhyllUse = toCopy.choosePhyllUse;
    minDL = toCopy.minDL;
    pFLLAnth = toCopy.pFLLAnth;
    dcd = toCopy.dcd;
    dgf = toCopy.dgf;
    degfm = toCopy.degfm;
    ignoreGrainMaturation = toCopy.ignoreGrainMaturation;
    pHEADANTH = toCopy.pHEADANTH;
    sLDL = toCopy.sLDL;
    sowingDay = toCopy.sowingDay;
    sowingDate = toCopy.sowingDate;
    sDws = toCopy.sDws;
    sDsa_nh = toCopy.sDsa_nh;
    der = toCopy.der;
    targetFertileShoot = toCopy.targetFertileShoot;
    dse = toCopy.dse;
    slopeTSFLN = toCopy.slopeTSFLN;
    intTSFLN = toCopy.intTSFLN;
}


void PhenologyComponent::Init(PhenologyState& s, PhenologyState& s1, PhenologyRate& r, PhenologyAuxiliary& a)
{
    s.calendarMoments.push_back("Sowing");
    s.calendarCumuls.push_back(0.0d);
    s.calendarDates.push_back(sowingDate);
    s.minFinalNumber = 5.5d;
}
